src/array_api_stubs/_draft/set_functions.py

```diff
@@ -1,7 +1,51 @@
-__all__ = ["unique_all", "unique_counts", "unique_inverse", "unique_values"]
+__all__ = ["isin", "unique_all", "unique_counts", "unique_inverse", "unique_values"]
 
 
-from ._types import Tuple, array
+from ._types import Tuple, Union, array
+
+
+def isin(
+    x1: Union[array, int, float, complex, bool],
+    x2: Union[array, int, float, complex, bool],
+    /,
+    *,
+    invert: bool = False,
+) -> array:
+    """
+    Tests whether each element in ``x1`` is in ``x2``.
+
+    Parameters
+    ----------
+    x1: Union[array, int, float, complex, bool]
+        first input array. **May** have any data type.
+    x2: Union[array, int, float, complex, bool]
+        second input array. **May** have any data type.
+    invert: bool
+        boolean indicating whether to invert the test criterion. If ``True``, the function **must** test whether each element in ``x1`` is *not* in ``x2``. If ``False``, the function **must** test whether each element in ``x1`` is in ``x2``. Default: ``False``.
```

Comment:
So if `isin(x1, x2, invert=True)` is exactly equivalent to `logical_not(isin(x1, x2))`, we could drop the argument completely.

Reply:
I recognize that the performance optimization argument I just stated is a bit tenuous here, given that the OP explicitly advocates against including `assume_unique`, but the difference between `invert` and `assume_unique` is that the former conveys semantic meaning, whereas the latter is primarily about making implementer lives easier.